scripts/grading/grade_submission.py

```
@@ -689,9 +859,19 @@ def read_text_file(path: Path | None) -> str:
     if suffix == ".pdf":
-        # Prefer pypdf (already in requirements)
+        try:
+            import fitz  # PyMuPDF
+            with fitz.open(str(path)) as doc:
+                parts = []
+                for page in doc:
+                    t = (page.get_text() or "").strip()
+                    if t:
+                        parts.append(t)
+            return "\n\n".join(parts).strip()
+        except Exception as exc:
```

Ruff (0.15.11)

[warning] Do not catch blind exception: `Exception`

(BLE001)